File: medplum-java/medplum-model/src/main/java/com/fhir/Contract_Offer.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableContract_Offer.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Contract_Offer {
  Optional<List<CodeableConcept>> decisionMode();

  Optional<List<Identifier>> identifier();

  Optional<List<String>> linkId();

  Optional<Reference> topic();

  Optional<String> text();

  Optional<List<Contract_Party>> party();

  Optional<CodeableConcept> type();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> decision();

  Optional<String> id();

  Optional<List<Contract_Answer>> answer();

  Optional<List<unsignedInt>> securityLabelNumber();

  Optional<List<Extension>> extension();

  static ImmutableContract_Offer.Builder builder() {
    return ImmutableContract_Offer.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Contract_Offer.ts
import { CodeableConcept } from "./CodeableConcept";
import { Contract_Answer } from "./Contract_Answer";
import { Contract_Party } from "./Contract_Party";
import { Extension } from "./Extension";
import { Identifier } from "./Identifier";
import { Reference } from "./Reference";
import { unsignedInt } from "./unsignedInt";

export interface Contract_Offer {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    party: Contract_Party[] | null | undefined;
    topic: Reference | null | undefined;
    type: CodeableConcept | null | undefined;
    decision: CodeableConcept | null | undefined;
    decisionMode: CodeableConcept[] | null | undefined;
    answer: Contract_Answer[] | null | undefined;
    text: string | null | undefined;
    linkId: string[] | null | undefined;
    securityLabelNumber: unsignedInt[] | null | undefined;
}
